dashboard/test/factories/pd_factories.rb

```diff
@@ -49,6 +52,57 @@
         workshop.enrollments << build(:pd_enrollment, workshop: workshop, user: teacher)
       end
     end
+
+    after(:create) do |workshop, evaluator|
+      workshop.sessions.map(&:save)
+
+      evaluator.num_facilitators.times do
+        workshop.facilitators << (create :facilitator)
+      end
+
+      evaluator.num_completed_surveys.times do
+        enrollment = create :pd_enrollment, workshop: workshop
+        if workshop.teachercon?
+          survey_hash = build :pd_teachercon_survey_hash
+
+          Pd::TeacherconSurvey.facilitator_required_fields.each do |field|
+            survey_hash[field] = {}
+          end
+
+          survey_hash['whoFacilitated'] = workshop.facilitators.map(&:name)
+
+          workshop.facilitators.each do |facilitator|
+            Pd::TeacherconSurvey.facilitator_required_fields.each do |field|
+              survey_hash[field][facilitator.name] = 'Free response'
+            end
+          end
+
+          create :pd_teachercon_survey, pd_enrollment: enrollment, form_data: survey_hash.to_json
```

Comment:
I think you misunderstood my suggestion. I'm talking about encapsulation. The workflow here will remain the same.

This whole section from line 63 - 79 is creating a `pd_teachercon_survey` based on a set of facilitators. Logically that belongs in the `pd_teachercon_survey` factory so it can be used on its own. As a new trait or transient attribute that shouldn't affect other tests. Then this section, inside `pd_workshop` factory can simply call `create :pd_teachercon_survey` and pass in the list of facilitators.